nxdrive/gui/api.py

```diff
@@ -443,6 +444,50 @@ def default_server_url_value(self) -> str:
         """Make daily job better for our developers :)"""
         return getenv("NXDRIVE_TEST_NUXEO_URL", "")
 
+    @pyqtSlot(str, int, result=list)
+    def disk_space_info_to_width(self, uid: str, width: int) -> List[float]:
+        engine = self._manager.engines.get(uid)
+        if not engine:
+            return [0, 0, 0]
+
+        space = engine.disk_space
+        used_without_synced = space.used - space.synced
+
+        return [
+            space.free * width / space.total,
+            used_without_synced * width / space.total,
+            space.synced * width / space.total,
+        ]
+
+    @pyqtSlot(str, result=bool)
+    def free_disk_space_under_limit(self, uid: str) -> bool:
+        engine = self._manager.engines.get(uid)
+
+        free_space = engine.disk_space.free if engine else 0
+        return free_space < FREE_DISK_SPACE_LIMIT
+
+    @pyqtSlot(str, result=str)
+    def get_drive_disk_space(self, uid: str) -> str:
+        engine = self._manager.engines.get(uid)
+
+        synced = engine.disk_space.synced if engine else 0
+        return sizeof_fmt(synced, suffix=Translator.get("BYTE_ABBREV"))
+
+    @pyqtSlot(str, result=str)
+    def get_free_disk_space(self, uid: str) -> str:
+        engine = self._manager.engines.get(uid)
+
+        free_space = engine.disk_space.free if engine else 0
+        return sizeof_fmt(free_space, suffix=Translator.get("BYTE_ABBREV"))
+
+    @pyqtSlot(str, result=str)
+    def get_used_space_without_synced(self, uid: str) -> str:
+        engine = self._manager.engines.get(uid)
+
+        synced = engine.disk_space.synced if engine else 0
```

Comment:
Sotre the result to save twice calls: `space = engine.disk_space` and then work with `space`.